packages/react-router/lib/dom/ssr/routeModules.ts

```diff
@@ -18,19 +18,82 @@ export interface RouteModules {
   [routeId: string]: RouteModule | undefined;
 }
 
+/**
+ * The shape of a route module shipped to the client
+ */
 export interface RouteModule {
```

Comment:
This can be a `type` as well

```suggestion
export type RouteModule = {
```